test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -28,52 +27,66 @@ public class MsbuildIntegrationTestFixture : IDisposable
+            TestDotnetCli = Path.Combine(_cliDirectory, dotnetExecutableName);
 
             MsBuildSdksPath = Path.Combine(Directory.GetDirectories
                 (Path.Combine(_cliDirectory, "sdk"))
                 .First(), "Sdks");
+            _templateDirectory = TestDirectory.Create();
 
             _processEnvVars.Add("MSBuildSDKsPath", MsBuildSdksPath);
             _processEnvVars.Add("UseSharedCompilation", "false");
             _processEnvVars.Add("DOTNET_MULTILEVEL_LOOKUP", "0");
             _processEnvVars.Add("MSBUILDDISABLENODEREUSE ", "true");
-            // We do this here so that dotnet new will extract all the packages on the first run on the machine.
-            InitDotnetNewToExtractPackages();
         }
 
-        private void InitDotnetNewToExtractPackages()
+        /// <summary>
+        /// Creates a new dotnet project of the specified type. Note that restore/build are not run when this command is invoked. 
+        /// That is because the project generation is cached.
+        /// </summary>
+        internal void CreateDotnetNewProject(string solutionRoot, string projectName, string args = "console", int timeOut = 60000)
         {
-            using (var testDirectory = TestDirectory.Create())
+            args = args.Trim();
+            var workingDirectory = Path.Combine(solutionRoot, projectName);
+            if (!Directory.Exists(workingDirectory))
             {
-                var projectName = "ClassLibrary1";
-                CreateDotnetNewProject(testDirectory.Path, projectName, " classlib", timeOut: 300000);
+                Directory.CreateDirectory(workingDirectory);
             }
+            var templateDirectory = Path.Combine(_templateDirectory.Path, args);
 
-            using (var testDirectory = TestDirectory.Create())
+            if (Directory.Exists(templateDirectory))
+            {
+                CopyFromTemplate(projectName, args, workingDirectory, templateDirectory);
+            }
+            else
             {
-                var projectName = "ConsoleApp1";
-                CreateDotnetNewProject(testDirectory.Path, projectName, " console", timeOut: 300000);
+                Directory.CreateDirectory(templateDirectory);
+
+                var result = CommandRunner.Run(TestDotnetCli,
+                    templateDirectory,
+                    $"new {args}",
+                    waitForExit: true,
+                    timeOutInMilliseconds: timeOut,
+                    environmentVariables: _processEnvVars);
+                // Delete the obj directory because it contains assets generated by running restore at dotnet new <template> time. 
+                // These are not relevant when the project is renamed 
+                Directory.Delete(Path.Combine(templateDirectory, "obj"), true);
+                CopyFromTemplate(projectName, args, workingDirectory, templateDirectory);
```

Comment:
Shouldn't this statement be after the assertions below?  No need to copy if the command failed.

Also, you can just call this statement once:
```C#
if (!Directory.Exists(templateDirectory))
{
    ...
}

CopyFromTemplate(projectName, args, workingDirectory, templateDirectory);